### PR TITLE
[FakeSelectInputSpirit] Better checking of multiple select's changed indexes

### PR DESCRIPTION
**Pull request by sampi**
@wiredearp @zdlm

Fixes issue #622

### Fixes
* [FakeSelectInputSpirit] Better checking of multiple select's changed indexes

The existing code iterated through the list of `<option selected></option>`'s and matched them against the list of already existing selected options'. If it would find something that exists in the new, but not the old then it triggered the internal change event to update the status.
This method didn't work when the new indexes are a subset of the old indexes.

The new code uses a symmetric diff algorithm to check if any of the two lists have values in them that are not existing in the other and will trigger the internal change event whenever this is the case.

### Fun fact
* Selecting `<option/>`'s using JQuery works only by using `.attr()`, but unselecting works both with `.attr()` & `.prop()`.
You can read more about this quasi-strange behavior here: http://api.jquery.com/attr/#svg-support

### CODE REVIEW

**Comment by sampi** [comment]
This is not a pretty page, so I set it as `class="hidden"`

screen shot 2018-07-16 at 15 20 51

File: src/spiritual/spiritual-gui/gui@wunderbyte.com/polyfills.js [hunk 1]
```diff
@@ -88,6 +88,24 @@
 					}
 				}
 				return undefined;
+			},
+			diffSymmetric: function diffSymmetric(otherArr) {
```
Comment (wiredearp) [inline src/spiritual/spiritual-gui/gui@wunderbyte.com/polyfills.js]: Although perhaps not a polyfill, I guess, but yeah.

Comment (sampi) [reply src/spiritual/spiritual-gui/gui@wunderbyte.com/polyfills.js]: where would you put it?

File: spec/runtime/spirits/forms-gui@tradeshift.com/ts.ui.SelectSpirit.spec.js [hunk 2]
```diff
@@ -7,6 +7,18 @@ describe('ts.ui.SelectSpirit', function likethis() {
 		'</select>'
 	].join('\n');
 
+	var MARKUP_MULTI = function(id, selectedArr) {
+		return [
+			'<div id="' + id + '">',
+			'<select data-ts="Select" multiple>',
+			'<option value="1" ' + (selectedArr[0] ? 'selected' : '') + '>One</option>',
```
Comment (wiredearp) [inline spec/runtime/spirits/forms-gui@tradeshift.com/ts.ui.SelectSpirit.spec.js]: Never forget!

**Pull request merged**